library/std/tests/time.rs

```diff
@@ -227,3 +227,29 @@ fn big_math() {
     check(instant.checked_add(Duration::from_secs(100)), Instant::checked_sub);
     check(instant.checked_add(Duration::from_secs(i64::MAX as _)), Instant::checked_sub);
 }
+
+#[test]
+#[cfg(unix)]
+fn system_time_extreme_values_regression() {
+    // Test for regression in SystemTime comparison with extreme values
+    // This test covers the bug introduced in PR #144519 where integer overflow
+    // in the comparison logic caused incorrect results when dealing with times
+    // near i64::MIN and i64::MAX.
+    //
+    // This is the exact test case from GitHub issue #146228
+    let t = SystemTime::UNIX_EPOCH;
+    let early = t - (Duration::from_secs(i64::MAX as u64 + 1));
+    let later = t + (Duration::from_secs(i64::MAX as u64) + Duration::from_nanos(999_999_999));
```

Comment:
Offset of the SystemTime by a Duration is not guaranteed to give a sensible result and I do not believe `cfg(unix)` is sufficient for this.

Reply:
In current std implementation, this operation supposed to succeed, because, `SystemTime` is defined using `Timespec` that has `i64` seconds part:

https://github.com/rust-lang/rust/blob/7ad23f43a225546c095123de52cc07d8719f8e2b/library/std/src/sys/pal/mod.rs#L27-L31

https://github.com/rust-lang/rust/blob/7ad23f43a225546c095123de52cc07d8719f8e2b/library/std/src/sys/pal/unix/time.rs#L20-L29

Rust std may want to reduce the range of `SystemTime` or `Timespec` one day, but I don't think this is likely, and if that happens, the test can be changed.